pkg/service/one2onerestore/model.go

```diff
@@ -49,9 +49,41 @@ func (t *Target) validateProperties() error {
 	if t.SourceClusterID == uuid.Nil {
 		return errors.New("source cluster id is empty")
 	}
-	if len(t.NodesMapping) == 0 {
-		return errors.New("nodes mapping is empty")
+	if err := validateNodesMapping(t.NodesMapping); err != nil {
+		return errors.Wrap(err, "nodes mapping")
 	}
+	return nil
+}
+
+func validateNodesMapping(nodesMapping []nodeMapping) error {
+	if len(nodesMapping) == 0 {
+		return errors.New("empty")
+	}
+
+	var (
+		sourceDCCount = map[string]int{}
+		targetDCCount = map[string]int{}
+
+		sourceRackCount = map[string]int{}
+		targetRackCount = map[string]int{}
+	)
 
+	for _, nodeMapping := range nodesMapping {
+		s, t := nodeMapping.Source, nodeMapping.Target
+
+		sourceDCCount[s.DC]++
+		targetDCCount[t.DC]++
+
+		sourceRackCount[s.DC+s.Rack]++
+		targetRackCount[t.DC+t.Rack]++
+
+		if sourceDCCount[s.DC] != targetDCCount[t.DC] {
+			return errors.Errorf("source and target clusters has different number of nodes per DC")
+		}
+
+		if sourceRackCount[s.DC+s.Rack] != targetRackCount[t.DC+t.Rack] {
+			return errors.Errorf("source and target clusters has different number of nodes per Rack")
+		}
```

Comment:
I would say that what we are trying to validate here is that mapping between src and dst DCs is 1-1.
So it's not just about having the same number of nodes per DC, e.g.:
- dc1:nodeX -> dc3:nodeX
- dc1:nodeX -> dc4:nodeX
- dc2:nodeX -> dc4:nodeX

where nodeX is some new unique node each time. Here we have the same number of nodes per DC (at least in my understanding), but we don't have the 1-1 mapping, which is important.

The validation code should work, because it checks the DC size after each iteration (although it isn't obvious, because the code also kind of focuses on the DC size, not the 1-1 mapping), but the error message should focus on the 1-1 mapping, not just the DC size.

The same goes for rack error message.

Reply:
Indeed, @VAveryanov8 you can extend your test with 
```
		{
			name: "test",
			nodesMapping: []nodeMapping{
				{Source: node{DC: "dc1", Rack: "rack1", HostID: "h1"}, Target: node{DC: "dc2", Rack: "rack1", HostID: "host1"}},
				{Source: node{DC: "dc1", Rack: "rack2", HostID: "h2"}, Target: node{DC: "dc2", Rack: "rack2", HostID: "host3"}},
				{Source: node{DC: "dc1", Rack: "rack2", HostID: "h3"}, Target: node{DC: "dc2", Rack: "rack1", HostID: "host2"}},
			},
			expectedErr: true,
		},
```

node from source_rack1 that has only 1 node is mapped to host from target rack that has 2 nodes.
It passes current validation, but shouldn't.